kratos/utilities/read_materials_utility.cpp

```diff
@@ -222,8 +222,10 @@ void ReadMaterialsUtility::AssignVariablesToProperty(
     // Add / override the values of material parameters in the p_properties
     if (MaterialData.Has("Variables")) {
         Parameters variables = MaterialData["Variables"];
-        for (auto iter = variables.begin(); iter != variables.end(); ++iter) {
-            const Parameters value = variables.GetValue(iter.name());
+        const Parameters variables_considered = FilterFailingVariablesIfRequired(variables, rProperty.Id());
```

Comment:
Ok, now it is more of my liking,

however what about making this function virtual (maybe called more genarally FilterVariables)

and implementing it directly in the derived class (the base class would do nothing)? that would be more flexible

Reply:
I thought about that, but the problem is the following. The implementation was already here, and I could not move the implementation to the MFront application because is not public yet. With the proposed one I only need to implement a true returning method, instead of the code I added and I would need to move in order to preserve this implementation

Reply:
An even better would be to essentially make virtual  all the functions in the ReadMaterialUtility (and adding an "empty" FilterVariables in the base class)

Even if i understand that you do not have access to the MFront app, you could simply pass the Filtering code to @thelfer (or if you want to do it yourself, adding a custom reader, IN A DIFFERENT FILE, which is what needs to be done anyway).
That way you don't need to modify the base class (which is used by everyone)

Reply:
Okay